File: Wildcat.Http/Client/Music/Block/Albums.js
(function() {
  var __hasProp = {}.hasOwnProperty,
    __extends = function(child, parent) { for (var key in parent) { if (__hasProp.call(parent, key)) child[key] = parent[key]; } function ctor() { this.constructor = child; } ctor.prototype = parent.prototype; child.prototype = new ctor; child.__super__ = parent.prototype; return child; };

  namespace("Music");

  namespace("Music.Block");

  Music.Block.Albums = (function(_super) {

    __extends(Albums, _super);

    Albums.name = 'Albums';

    function Albums() {
      return Albums.__super__.constructor.apply(this, arguments);
    }

    Albums.prototype.OnLoadData = function(data) {
      this.model = data;
      return this.render();
    };

    Albums.prototype.render = function() {
      return this.container.text("Это Альбомы");
    };

    return Albums;

  })(Wildcat.Block);

}).call(this);

File: Wildcat.Music/Block/Albums.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Newtonsoft.Json.Linq;

namespace Music.Block
{
    public class Albums : Wildcat.DB.System.Block
    {
        public new Collection.Album Model;

        public override void OnInitSmall(Wildcat.DB.System.Block block)
        {
            var parent = (Block.Artist)Parent;
            Model = parent.Model.GetAlbums();
            State = BlockState.Small;
        }

        public void OnLoadData(JObject data)
        {
            if (Model != null)
            {
                var jObj = new JObject();
                jObj.Add("collection", JArray.FromObject(Model));

                Console.WriteLine(jObj.ToString());
                SendToClient("OnLoadData", jObj);
            }
        }
    }
}
